Jenkinsfile

```diff
@@ -26,17 +27,14 @@ node("k8s") {
             // anything to include it anyway.
             withRepoKey {
                 k8s_contexts.each { cluster ->
-                    template(
-                        basedir: "${env.WORKSPACE}/${cluster}",
-                        cluster: cluster
-                    )
-                    sh("ls ${env.WORKSPACE}/${cluster}/tmp-k8s/")                 
+                    template(cluster: cluster)
+                    sh("kubectl --context ${cluster} apply -f ${env.WORKSPACE}/${cluster}/tmp-k8s")                 
                 }
             }
         }
     }
 
-    if (env.BRANCH_NAME.equals("main")) {
+    else if (env.BRANCH_NAME.equals("staging")) {
         stage("Deploy") {
```

Comment:
You can remove this entire other case for staging. Please include the first block that's on main in its place. That block will test templating but not apply if we are on any branch besides main.

Reply:
```if (!env.BRANCH_NAME.equals("main"))```